05.ls/ls1.rb

```diff
@@ -1,30 +1,128 @@
 # frozen_string_literal: true
 
 # !/usr/bin/env ruby
+require 'optparse'
+require 'etc'
+
 NUMBER_OF_COLUMNS = 3
+FILE_TYPE = {
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '-',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+PERMISSION_PATTERN = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
```

Comment:
`FILE_TYPE`とか`PERMISSION_PATTERN`がちょっとゴツいので、改行を開けた方が見やすいかなと思いました。
```suggestion

FILE_TYPE = {
  'fifo' => 'p',
  'characterSpecial' => 'c',
  'directory' => 'd',
  'blockSpecial' => 'b',
  'file' => '-',
  'link' => 'l',
  'socket' => 's'
}.freeze

PERMISSION_PATTERN = {
  '0' => '---',
  '1' => '--x',
  '2' => '-w-',
  '3' => '-wx',
  '4' => 'r--',
  '5' => 'r-x',
  '6' => 'rw-',
  '7' => 'rwx'
}.freeze
```